### PR TITLE
fix: Duplicated calling system messages (AR-2744)

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Issue 1 : When you start a 1:1 call and end it without accepting on the other side, “You called” system message is displayed twice

Issue 2: When you reject an incoming call and the caller terminates the call without being accepted, “Someone called” is displayed twice

### Causes (Optional)

We were updating the call two times when the call is ended or rejected: The first time was in `OnCloseCall` callback and the second was in `endCall()`/`rejectCall()` method
Also we were not checking the `reason` field in `OnCloseCall` callback

### Solutions

Remove duplicated code and Implement missing parts
Also some clean up is performed: Removed `OnMissedCall` callback since it never occurs 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Issue 1:
- Start a 1:1 call and end it without accepting on the other side.
Issue 2:
- Reject an incoming 1:1 call and wait until the caller terminates it (The call should not be accepted anywhere)

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt [hunk 1]
```diff
@@ -235,13 +226,6 @@ class CallManagerImpl internal constructor(
 
     override suspend fun rejectCall(conversationId: ConversationId) = withCalling {
         callingLogger.d("[$TAG][rejectCall] -> ConversationId: [$conversationId]")
-        val conversationType = callRepository.getLastCallConversationTypeByConversationId(conversationId = conversationId)
-
-        callingLogger.d("[$TAG][rejectCall] -> ConversationType: [$conversationType]")
-        callRepository.updateCallStatusById(
-            conversationIdString = conversationId.toString(),
-            status = if (conversationType == Conversation.Type.GROUP) CallStatus.STILL_ONGOING else CallStatus.CLOSED
-        )
```
Comment (alexandreferris) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:244]: By removing the update of call status here, the user will then be waiting for this flow:

`user is in a call -> user ends the call -> [request to UseCase -> Repository -> Network/AVS] -> [response from AVS -> Repository -> UseCase] -> User is alerted and only then the screen is closed`

By having the call to update the call status here directly, we don't need to wait on Network/AVS response and only then closing the screen, so the user doesn't need to wait the whole process for the call screen to close.

By removing it did you see any drastic change on the wait time for the call screen to be closed/dismissed?

File: logic/src/jvmMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt [hunk 2]
```diff
@@ -236,14 +227,6 @@ class CallManagerImpl internal constructor(
...
     override suspend fun rejectCall(conversationId: ConversationId) = withCalling {
         callingLogger.d("[$TAG][rejectCall] -> ConversationId: [$conversationId]")
-        val conversationType = callRepository.getLastCallConversationTypeByConversationId(conversationId = conversationId)
-
-        callingLogger.d("[$TAG][rejectCall] -> ConversationType: [$conversationType]")
-        callRepository.updateCallStatusById(
-            conversationIdString = conversationId.toString(),
-            status = if (conversationType == Conversation.Type.GROUP) CallStatus.STILL_ONGOING else CallStatus.CLOSED
-        )
-
```
Comment (alexandreferris) [inline logic/src/jvmMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:246]: By removing the update of call status here, the user will then be waiting for this flow:

`user is in a call -> user ends the call -> [request to UseCase -> Repository -> Network/AVS] -> [response from AVS -> Repository -> UseCase] -> User is alerted and only then the screen is closed`

By having the call to update the call status here directly, we don't need to wait on Network/AVS response and only then closing the screen, so the user doesn't need to wait the whole process for the call screen to close.

By removing it did you see any drastic change on the wait time for the call screen to be closed/dismissed?

**Review by alexandreferris — COMMENTED** [review]
Great work on handling the duplicated messages!

I just have some concerns on the changes and would like to get some input on the reasons, but mostly understand it

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt [hunk 3]
```diff
@@ -218,14 +217,6 @@ class CallManagerImpl internal constructor(
...
     override suspend fun endCall(conversationId: ConversationId) = withCalling {
         callingLogger.d("[$TAG][endCall] -> ConversationId: [$conversationId]")
-        val conversationType = callRepository.getLastCallConversationTypeByConversationId(conversationId = conversationId)
-
-        callingLogger.d("[$TAG][endCall] -> ConversationType: [$conversationType]")
-        callRepository.updateCallStatusById(
-            conversationIdString = conversationId.toString(),
-            status = if (conversationType == Conversation.Type.GROUP) CallStatus.STILL_ONGOING else CallStatus.CLOSED
-        )
-
```
[the remark this answers is not in the document]
Comment (ohassine) [reply logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:228]: Good point, slightly visible delay but we can handle this on client side

**Pull request merged**